File: WebService-Korpus/src/main/java/org/alex73/korpus/server/data/WordResult.java
package org.alex73.korpus.server.data;

import org.alex73.korpus.text.structure.corpus.Word;

@SuppressWarnings("serial")
public class WordResult extends Word {
    /** True if word is requested by user, i.e. should be marked in output. */
    public Boolean requestedWord;

    public WordResult(Word w) {
        this.word = w.word;
        this.tail = w.tail;
        this.type = w.type;
    }

    @Override
    public String toString() {
        return word + tail;
    }
}

File: Web/src/main/resources/ts/dto/WordResult.ts
class WordResult {
  public word: string;
  public tail: string;
  public requestedWord: boolean;

  constructor(private text: string) {
    this.word = text;
  }
  getOutput(): string {
    return this.word;
  }
}

class Sentence {
  public words: WordResult[];
}

// See org.alex73.korpus.text.structure.corpus.Paragraph.
class Paragraph {
  public page: number;
  public lang: string;
  public previewLink: string;
  public sourceLink: string;
  public sentences: Sentence[];
}
